### DISCUSSION TITLE
RabbitMQ Scaler should look for hostFromEnv in all Containers

### DISCUSSION THREAD

**Discussion by oksiquatzel** [kedacore/keda#2046] [Q&A / Need Help] [answered]
### Proposal

RabbitMQ Scaler should look in all Containers of the scaledTarget for the ENV Variable, or there should be a way to define in witch container the ENV is specified.

### Use-Case

sometimes you have multiple Containers in one deployment with different ENVs defined. 
For example a RabbitMQ Worker Container and a Database Balance Container. Then we like to define the ENV for keda only in the Worker Container. Currently we need to define the ENV in the first Container defined in the Deployment, for Keda to work.  

### Anything else?

We use 
keda 2.4
K8S: v1.21

**Comment by tomkerkhove** [comment, marked as answer]
You can use `envSourceContainerName` in your `ScaledObject` to define this:
https://keda.sh/docs/2.4/concepts/scaling-deployments/#details

**Reply by oksiquatzel** [reply]
Oh really. Thats nice, i could not find it when searching for containers, on https://keda.sh 
Sorry for the wrong Ticket 
Thank you and keep on your really good work

**Reply by tomkerkhove** [reply]
Happy to help!
